handlers/handlers.go

```diff
@@ -40,68 +39,60 @@ func setStatusCode(req *http.Request, w http.ResponseWriter, err error) {
 	w.WriteHeader(status)
 }
 
-func forwardFlorenceTokenIfRequired(req *http.Request) *http.Request {
-	if len(req.Header.Get(common.FlorenceHeaderKey)) > 0 {
-		ctx := common.SetFlorenceIdentity(req.Context(), req.Header.Get(common.FlorenceHeaderKey))
-		return req.WithContext(ctx)
-	}
-	return req
-}
-
-//GeographyRender ...
-func GeographyRender(rend RenderClient) http.HandlerFunc {
+//HomepageRender gets geography data from the code-list-api and formats for rendering
+func HomepageRender(rend RenderClient, cli *codelist.Client) http.HandlerFunc {
 
 	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
 		var page geographyHomepage.Page
 
-		resp, err := http.Get(`https://api.dev.cmd.onsdigital.co.uk/v1/code-lists` + localAuthority)
-		if err != nil {
-			setStatusCode(req, w, err)
-			return
-		}
-		b, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			setStatusCode(req, w, err)
-			return
-		}
-		var codelistresults models.CodeListResults
-		err = json.Unmarshal(b, &codelistresults)
+		codeListResults, err := cli.GetCodelistData()
 		if err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "error geting code-lists data for geography"), nil)
 			setStatusCode(req, w, err)
 			return
 		}
-		var codelist models.CodeList
-		err = json.Unmarshal(b, &codelist)
-		if err != nil {
-			setStatusCode(req, w, err)
-			return
-		}
-
-		geographyTypes := ""
-		for i := range codelistresults.Items {
-			log.Debug("test", log.Data{
-				"geographyTypesTest": codelistresults.Items[i],
-			})
-
-			geographyTypes = geographyTypes + codelistresults.Items[i].Name
-
-		}
 
-		page.Data.AreaTypes = []geographyHomepage.AreaType{
-			{Name: "Countries"},
-			{Name: "Regions"},
-			{Name: geographyTypes},
+		var types []geographyHomepage.Items
+		var wg sync.WaitGroup
+		var mutex = &sync.Mutex{}
+		for _, v := range codeListResults.Items {
+			wg.Add(1)
+			go func(codeListResults models.CodeListResults, cli *codelist.Client, v models.CodeList) {
+
+				typesID := v.Links.Self.ID
+				editionsListResults, err := cli.GetEditionslistData(v.Links.Editions.Href)
+				if err != nil {
+					return
+				}
+
+				if len(editionsListResults.Items) > 0 && editionsListResults.Items[0].Label != "" {
+					mutex.Lock()
+					types = append(types, geographyHomepage.Items{
+						Label: editionsListResults.Items[0].Label,
+						ID:    typesID,
+					})
+					mutex.Unlock()
+				}
+
+				wg.Done()
+				return
+			}(codeListResults, cli, v)
 		}
+		wg.Wait()
 
+		page.Data.Items = types
```

Comment:
observation rather than asking for a change - we could just initialise `page.Data.Items` and append to that rather than creating `types` and assigning it later, no functional difference though, just a tiny bit less code